lib/serializer.js

```diff
@@ -27,7 +27,7 @@ module.exports = class Serializer {
       if (i === Infinity || i === -Infinity) {
         throw new Error(`The value "${i}" cannot be converted to an integer.`)
       }
-      if ((i | 0) === i) {
+      if (Number.isInteger(i)) {
```

Comment:
Shouldn't there be a test that exposes why this method is used over the other (which I agree this is the better method)? Although, I'm not sure how to represent a `BigInt` to trigger this.

Reply:
bigint would jump to the branch in line 39. I added a test, which theoretically should jump to that branch.

Reply:
It is not breaking existing code, because it would anyway be casted to an integer.

Reply:
Test clears the bar on my system 👍